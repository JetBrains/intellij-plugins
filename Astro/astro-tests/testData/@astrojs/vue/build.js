const path = require('path');
const fs = require('fs');
const esbuild = require('esbuild');

const targetPackage = '@astrojs/vue';

const pkgPath = require.resolve(`./node_modules/${targetPackage}/package.json`);
const theirPkg = require(pkgPath);
const pkgDir = path.dirname(pkgPath);

const entries = {
  './editor': path.join(pkgDir, 'dist/editor.cjs'),
};

const outDir = path.resolve(__dirname, 'dist');
fs.rmSync(outDir, {recursive: true, force: true});
fs.mkdirSync(outDir, {recursive: true});

const externals = [
  'twing', 'twig', 'velocityjs', 'dustjs-linkedin', 'atpl', 'liquor',
  'hamljs', 'templayed', 'handlebars', 'mustache', 'just',
  'haml', 'haml-coffee', 'hamlet', 'whiskers', 'hogan.js', 'eco', 'ejs',
  'jade', 'pug', 'jazz', 'jqtpl', 'swig', 'walrus', 'squirrelly', 'eta',
  'nunjucks', 'underscore', 'lodash', 'marko', 'plates', 'react', 'react-dom',
  'dot', 'ect', 'tinyliquid', 'ractive', 'coffee-script', 'mote', 'toffee', 'bracket-template',
  'htmling', 'babel-core', 'vash', 'slm', 'teacup/lib/express'
];

// Plugin: treat any native addon (*.node) as external
function externalizeNativeNode() {
  return {
    name: 'externalize-native-node',
    setup(build) {
      build.onResolve({filter: /\.node$/}, (args) => ({path: args.path, external: true}));
    },
  };
}

async function buildEntry(entryPath, outFileName, format) {
  const outfile = path.join(outDir, outFileName);
  await esbuild.build({
    entryPoints: [entryPath],
    outfile,
    bundle: true,
    platform: 'node',
    target: 'node18',
    format, // 'esm' or 'cjs'
    minify: process.argv.includes('--minify'),
    sourcemap: false,
    define: {'process.env.NODE_ENV': '"production"'},
    external: externals,
    plugins: [externalizeNativeNode()],
  });
  return outfile;
}

async function buildAll() {
  console.log(`[build] Bundling ${targetPackage}@${theirPkg.version}`);

  const builtFiles = [];

  for (const [key, entryPath] of Object.entries(entries)) {
    if (key === '.') {
      console.log(` - ${key} (esm)`);
      builtFiles.push({key, format: 'esm', outfile: await buildEntry(entryPath, 'index.mjs', 'esm')});

      console.log(` - ${key} (cjs)`);
      builtFiles.push({key, format: 'cjs', outfile: await buildEntry(entryPath, 'index.cjs', 'cjs')});
      continue;
    }

    if (key === './client.js') {
      console.log(` - ${key} (esm) → client.svelte.mjs`);
      builtFiles.push({
        key,
        format: 'esm',
        outfile: await buildEntry(entryPath, 'client.svelte.js', 'esm'),
      });
      continue;
    }

    if (key === './server.js') {
      console.log(` - ${key} (esm) → server.mjs`);
      builtFiles.push({
        key,
        format: 'esm',
        outfile: await buildEntry(entryPath, 'server.js', 'esm'),
      });
      continue;
    }

    if (key === './editor') {
      console.log(` - ${key} (cjs) → editor.cjs`);
      builtFiles.push({
        key,
        format: 'cjs',
        outfile: await buildEntry(entryPath, 'editor.cjs', 'cjs'),
      });
      continue;
    }
  }
}

buildAll().catch((err) => {
  console.error(err);
  process.exit(1);
});