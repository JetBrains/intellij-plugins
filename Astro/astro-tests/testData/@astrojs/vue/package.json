{
  "name": "@astrojs/svelte",
  "version": "7.2.0",
  "private": true,
  "devDependencies": {
    "esbuild": "^0.20.0",
    "@astrojs/vue": "5.1.1"
  },
  "scripts": {
    "build": "node build"
  }
}