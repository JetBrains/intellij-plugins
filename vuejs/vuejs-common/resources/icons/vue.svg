<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
  <g fill="none" fill-rule="evenodd" transform="translate(1.09 2)">
    <polygon fill="#42B883" points="8.479 0 6.888 2.755 5.298 0 0 0 6.888 11.931 13.777 0"/>
    <polygon fill="#35495E" points="8.479 0 6.888 2.755 5.298 0 2.755 0 6.888 7.159 11.022 0 8.479 0"/>
  </g>
</svg>
