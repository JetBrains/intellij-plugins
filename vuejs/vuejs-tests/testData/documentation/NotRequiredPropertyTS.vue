<template>
  <div class="hello">
    <h1>{{ m<caret>sg }}</h1>
  </div>
</template>

<script lang="ts">
import {defineComponent, PropType} from 'vue';

export default defineComponent({
  name: 'HelloWorld',
  props: {
    msg: {
      type: String as PropType<string>
    }
  },
  mounted() {
    console.log(this.msg)
  }
});
</script>