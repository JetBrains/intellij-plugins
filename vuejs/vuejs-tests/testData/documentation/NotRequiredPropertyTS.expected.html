<div class='definition'><pre><span style="">msg:&#32;</span><span style="color:#000080;">string</span></pre></div><table class='sections'><tr><td valign='top'><icon src='AllIcons.FileTypes.JavaScript'/>&nbsp;NotRequiredPropertyTS.vue</td></table>